jade/trusted_clients.py

```diff
@@ -13,9 +15,21 @@ def authorize_and_decode(self, request):
         auth_key = request.values['auth_key']
         encoded_values = request.values['encoded_values']
 
-        return json2multidict(jwt.decode(
-            encoded_values, self.key_secrets[auth_key],
-            algorithms=[HASH_ALGORITHM]))
+        try:
+            return json2multidict(jwt.decode(
+                encoded_values, self.key_secrets[auth_key],
+                algorithms=[HASH_ALGORITHM]))
+        except (jwt.exceptions.InvalidTokenError,
```

Comment:
This was a pain because not all JWT errors inherit from a common base error type.  

```
$ python
Python 3.5.1+ (default, Mar 30 2016, 22:46:26) 
[GCC 5.3.1 20160330] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> error_types = (RuntimeError, ValueError)
>>> try:
...   raise RuntimeError()
... except error_types as e:
...   print("It worked")
... except Exception as e:
...   print("It didn't work")
... 
It worked
```

I can't believe that worked.  :D

Reply:
lol.  Python and Ruby are amazing in that everything is a first-class object.